cmd/operator-sdk/new/cmd.go

```diff
@@ -100,8 +100,13 @@ generates a default directory layout based on the input <project-name>.
 	newCmd.Flags().StringVar(&repo, "repo", "",
 		"Project repository path for Go operators. Used as the project's Go import path. This must be set if "+
 			"outside of $GOPATH/src (e.g. github.com/example-inc/my-operator)")
+
+	//todo: remove before 1.0.0
 	newCmd.Flags().BoolVar(&gitInit, "git-init", false,
 		"Initialize the project directory as a git repository (default false)")
+	_ = newCmd.Flags().MarkDeprecated("git-init", "This flag will be removed soon. "+
```

Comment:
you can also extract that to  an external file I think, just pass the newcmd as an argument